src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -539,15 +539,20 @@ int wm_checks_package_vulnerability(char *version_a, const char *operation, cons
         package_feed_version->version = cversion_it;
         package_feed_version->revision = crelease_it;
 
-        if (!strncmp(operation, vu_package_comp[VU_COMP_L], strlen(vu_package_comp[VU_COMP_L]))) {
+        if ((strlen(operation) == strlen(vu_package_comp[VU_COMP_L])) && 
```

Comment:
This comparison was causing the vulnerability detector to never be able to make use of the types `LE` or `GE`. This was caused because of the usage of the function `strncmp` that compares `n` number of characters.
So, when `operation` was equals to `"less than or equal"`, in this first `if` statement, it was comparing that string with the nine characters of `"less than"`, resulting as `true`.

The same happened in the third if statement between `"greater than"` and `"greater than or equal"`.

Reply:
Good catch!

Reply:
By using `strcmp` instead of `strncmp` yo don't need to calculate their length manually.